packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -275,6 +275,16 @@ abstract class TextSelectionControls {
   /// Returns the size of the selection handle.
   Size getHandleSize(double textLineHeight);
 
+  /// Returns the offset from text selection handle to actual location in text line.
```

Comment:
what is `actual location in text line.` ?

Reading the these documentation, I can't tell the difference between `getHandleAnchor` and this method

Reply:
and this seems to only used when calculating the end handle?

Reply:
`getHandleAnchor` use to locate different style handle from same `handleLayerLink`.

![getHandle](https://user-images.githubusercontent.com/30322203/183801112-889336f3-14e0-4add-87e1-109987a05798.png)

Reply:
`getOffsetFromHandleToTextPosition` use to transform drag point offset to actual location. actual location mean offset `RenderEditable` ues to update selection.

![drag](https://user-images.githubusercontent.com/30322203/183801603-ff1bfdb1-560a-451c-a0cc-cf1bbe2e55d5.png)

before this PR，Material Handle and Cupertino Handle use same way to transfrom drag point offset to actual location, the offset is `Offset(0.0, -handleSize.height)`.Because Material Style Handle is locate below the text line,
this offset is useful to `RenderEditable` update selection. But Cupertino Style Handle is overlaps the text line,so it cause wrong selection update, selection always update to previous line. And `EditableText` don't allow `baseOffset` greater than `extendOffset`,it cuase `endHandle` drag not smooth.

And it not only used when calculating the end handle. start handle and end handle has same problem.you can find this bug in any `CupertinoTextField` with multiline.

Reply:
this is demo code：

```
import 'package:flutter/material.dart';
import 'package:flutter/cupertino.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {

  TextEditingController controller = TextEditingController(text: '''
  Two roads diverged in a yellow wood,
  And sorry I could not travel both
  And be one traveler, long I stood
  And looked down one as far as I could
  To where it bent in the undergrowth;
  ''');

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: CupertinoTextField(
          controller: controller,
          maxLines: null,
        ),
      ),
    );
  }
}
```

Reply:
this is reproduce video:

https://user-images.githubusercontent.com/30322203/183806830-e84c959a-07d0-450f-aa5e-294917bf8b6a.mp4

Reply:
After use different `getOffsetFromHandleToTextPosition ` to update selection:

https://user-images.githubusercontent.com/30322203/183832611-b7d20360-6712-4da9-b86e-2cf29c807059.mp4

Reply:
> what is `actual location in text line.` ?
> 
> Reading the these documentation, I can't tell the difference between `getHandleAnchor` and this method

actual location in text line  mean the point RenderEditable use to get TextPosition and update selection. I will call the point actual text positon in later discussion.

Reply:
> and this seems to only used when calculating the end handle?

No, start handle and end handle with cupertino style has same problem, selection always update to previous text line.And EditableText don't allow baseOffset greater than extendOffset,so it cuase endHandle drag not smooth.